digital_land/commands.py

```diff
@@ -531,6 +533,220 @@ def collection_add_source(entry, collection, endpoint_url, collection_dir):
+    # ===== FIRST VALIDATION BASED ON IMPORT.CSV INFO
+    # - License okay, url okay, date format valid, organistion exist
+
+    # read and process each record of the new endpoints csv at csv_file_path i.e import.csv
+    with open(csv_file_path) as new_endpoints_file:
+        reader = csv.DictReader(new_endpoints_file)
+        csv_columns = reader.fieldnames
+
+        # validate the columns
+        for expected_col in expected_cols:
+            if expected_col not in csv_columns:
+                raise Exception(f"required column ({expected_col}) not found in csv")
+
+        endpoints = []
+        for row in reader:
+            if row["licence"] == "" or row["licence"] not in valid_licenses:
+                if row["licence"] == "":
+                    raise ValueError("Licence is blank")
+                else:
+                    raise ValueError(
+                        f"Licence '{row['licence']}' is not a valid licence according to the specification."
+                    )
+            # check if urls are not blank and valid urls
+            is_url_valid(row["documentation-url"], "documentation-url")
+            is_url_valid(row["endpoint-url"], "endpoint_url")
+
+            # if there is no start-date, do we want to populate it with today's date?
+            if row["start-date"]:
+                try:
+                    given_start_date = row["start-date"]
+
+                    try:
+                        given_start_date = datetime.strptime(
+                            row["start-date"], "%Y-%m-%d"
+                        ).date()
+                    # need to raise ValueError here otherwise datetime will raise it's own error, not the clear format we want
+                    except ValueError:
+                        raise ValueError(
+                            f"start-date {given_start_date} must be format YYYY-MM-DD"
+                        )
+
+                    if given_start_date > datetime.today().date():
+                        raise ValueError(
+                            f"The start_date {given_start_date} cannot be in the future"
+                        )
+                except ValueError:
+                    raise
+                except Exception as e:
+                    raise ValueError(f"An unexpected error occured: {str(e)}")
+
+            if (
+                row["organisation"] == ""
+                or row["organisation"] not in valid_organisations
+            ):
+                if row["organisation"] == "":
+                    raise ValueError(
+                        f"The given organisation '{row['organisation']}' is blank"
+                    )
+                if row["organisation"] not in valid_organisations:
+                    raise ValueError(
+                        f"The given organisation '{row['organisation']}' is not in our valid organisations"
+                    )
+
+            # do we need a check to see if the given pipeline exists?
```

Comment:
I actually think this is a good idea just to catch typo's (or people putting in a collection name by mistake), and we could expand it to include checking whether the pipeline is in the collection they give us. This could help catch cases where people are reusing the `add-data` command but forget to change the collection name/collection-dir in the command
You guys have more experience in this so let me know if you think it would be a helpful check or not

Reply:
Checking if the given collection exists would be a great addition I believe and it should be easy enough to include this

Reply:
I think it will be useful to check if the given pipeline exists. 
Checking if someone has used collection name instead of pipeline could be tricky because collection and pipeline name could be the same.

Reply:
we'll catch people who give us a collection name instead of pipeline/dataset just by way of checking if the pipeline/dataset exists - it won't be perfect of course, (e.g. brownfield-land collection name and brownfield-land dataset) but there's not much we can do in those cases, and in those cases where they're the same so it doesn't matter anyway.

my suggestion was to check whether the pipeline/dataset they give us belongs to the collection they give us as an additional layer to prevent user error